querydsl-core/src/main/java/com/querydsl/core/types/dsl/CaseBuilder.java

```diff
@@ -186,7 +186,7 @@ public Initial(Predicate b) {
             return new Cases<A, SimpleExpression<A>>((Class)expr.getType()) {
```

Comment:
I think relaxing the type parameter in `Cases`' `type`,`createResult` and constructor to `Class<? extends A>` helps a lot too to eliminate these casts **and** `@SuppressWarnings`.